### PR TITLE
Maps

### PR DESCRIPTION
**Pull request by whmii**
https://youtu.be/oIIxlgcuQRU

### CODE REVIEW

File: source/stylesheets/refills/_flashes.scss [hunk 1]
```diff
@@ -1,40 +1,28 @@
 $base-spacing: 1.5em !default;
-$alert-color: #fff6bf !default;
-$error-color: #fbe3e4 !default;
-$notice-color: #e5edf8 !default;
-$success-color: #e6efc2 !default;
+$flashes: (
+  alert: #fff6bf !default,
```
Comment (whmii) [inline source/stylesheets/refills/_flashes.scss]: hmmm

File: source/stylesheets/refills/_badges.scss [hunk 2]
```diff
@@ -1,36 +1,20 @@
...
-$alert-color: #fff6bf !default;
-$error-color: #fbe3e4 !default;
-$notice-color: #e5edf8 !default;
-$success-color: #e6efc2 !default;
+$small-font-size: 0.75em;
+$badges: (
+  default: #999,
+  alert: #fff6bf,
+  error: #fbe3e4,
+  notice: #e5edf8,
```
Comment (whmii) [inline source/stylesheets/refills/_badges.scss]: lolololol

File: source/stylesheets/refills/_flashes.scss (hunk 1 above)
Comment (tysongach) [inline source/stylesheets/refills/_flashes.scss]: Interesting! It appears you can only set the `!default` flag on standard variables, and not within a map: http://sass-lang.com/documentation/file.SASS_REFERENCE.html#variable_defaults_

Comment (whmii) [inline source/stylesheets/refills/_flashes.scss]: haha, TIL

File: source/stylesheets/refills/_badges.scss [hunk 3]
```diff
@@ -1,36 +1,20 @@
-$badge-font-size: 0.75em !default;
-$medium-gray: #999 !default;
-$alert-color: #fff6bf !default;
-$error-color: #fbe3e4 !default;
-$notice-color: #e5edf8 !default;
-$success-color: #e6efc2 !default;
+$small-font-size: 0.75em;
+$badges: (
+  default: #999,
```
Comment (Magnus-G) [inline source/stylesheets/refills/_badges.scss]: How come the `!default` is removed from this color definition? The idea was to be able to let Bitters override this one with the `$medium-gray` it defines, Bitters is used.

File: source/stylesheets/refills/_flashes.scss [hunk 4]
```diff
@@ -1,40 +1,28 @@
...
+);
 
-@mixin flash($color) {
-  background-color: $color;
-  color: darken($color, 60%);
-  display: block;
-  margin-bottom: $base-spacing / 2;
-  padding: $base-spacing / 2;
-  text-align: center;
+@each $flash-type, $color in $flashs {
```
Comment (tysongach) [inline source/stylesheets/refills/_flashes.scss]: `flashs` » `flashes`

File: source/stylesheets/refills/_flashes.scss (hunk 1 above)
Comment (tysongach) [inline source/stylesheets/refills/_flashes.scss]: So you can have a `!default` flag here. You just put it at the end, outside of the map:

```scss
$flashes: (
  alert: #fff6bf,
  error: #fbe3e4,
  notice: #e5edf8,
  success: #e6efc2,
) !default;
```

Though, why would we need to? These are encapsulated in a map now, and they would be copied in the source. So just edit the original map directly. If it was wrapped inside a library, the `!default` would be nice.

File: source/stylesheets/refills/_badges.scss (hunk 3 above)
Comment (tysongach) [inline source/stylesheets/refills/_badges.scss]: It’s a limitation of using Sass maps and I think we thought they couldn't be added…however, per conversation above, looks like you can have a `!default` flag on maps, it just goes once at the very end.

**Comment by tysongach** [comment]
@whmii What’s the status here? Mind tweaking per comments above, go through one more review, then get this merged?

**Comment by tysongach** [comment]
@whmii Ping-a-ling

**Comment by whmii** [comment]
Will fix

**Comment by tysongach** [comment]
@whmii :bell: friendly nudge :wink:

**Comment by tysongach** [comment]
@whmii 

giphy

**Comment by whmii** [comment]
ha, will fix

**Pull request merged**